### ISSUE TITLE
fetch mode specific parameters 

### ISSUE THREAD

**Issue by salarmehr** [phalcon/cphalcon#3191]
How should I pass each fetch mode specific parameters (http://php.net/manual/en/pdostatement.setfetchmode.php) in the following statement. Here I need to pass the colno parameter.

```
$this->di['db']->fetchOne('select role from user', Phalcon\Db::FETCH_COLUMN,[]);
```
Currently I get this error: SQLSTATE[HY000]: General error: fetch mode requires the colno argument

**Comment by olivier-monaco** [comment]
@phalcon In Phalcon 1.3.4, the class Phalcon\Db includes all FETCH_* constants from PDO but not in 2.0. Constant values in 2.0 are not the same as PDO like in 1.3.4. Is it a normal behavior?

Maybe Phalcon\Db\Adapter::fetch* methods implementation can be removed from this class and must be provided by concrete classes. Then Phalcon\Db\Adapter\Pdo can rely on underlined PDO fetch methods and then avoid to rewrite the same code as PDO (eg. current fetchColumn implementation).

And a fetchColumn method can be added to Phalcon\Db\ResultInterface and implemented in Phalcon\Db\Result\Pdo.

What do you think?

@salarmehr With Phalcon 1.3.4, you can really use FETCH_COLUMN. You can use fetchOne and than retrieves the first element from the returned array. As a note, if you really need only one column from your table, you must replace "*" with the column name for performance reasons.

**Comment by sergeyklay** [comment]
@olivier-monaco 
> In Phalcon 1.3.4, the class Phalcon\Db includes all FETCH_* constants from PDO but not in 2.0. 

https://github.com/phalcon/cphalcon/blob/2.0.0/phalcon/db.zep#L61 ?

**Comment by olivier-monaco** [comment]
@sergeyklay  I don't understand...

Phalcon 1.3.4
<pre>
	zend_declare_class_constant_long(phalcon_db_ce, SL("FETCH_USE_DEFAULT"), (long int)PDO_FETCH_USE_DEFAULT TSRMLS_CC);
	zend_declare_class_constant_long(phalcon_db_ce, SL("FETCH_LAZY"),        (long int)PDO_FETCH_LAZY TSRMLS_CC);
	zend_declare_class_constant_long(phalcon_db_ce, SL("FETCH_ASSOC"),       (long int)PDO_FETCH_ASSOC TSRMLS_CC);
	zend_declare_class_constant_long(phalcon_db_ce, SL("FETCH_NUM"),         (long int)PDO_FETCH_NUM TSRMLS_CC);
	zend_declare_class_constant_long(phalcon_db_ce, SL("FETCH_BOTH"),        (long int)PDO_FETCH_BOTH TSRMLS_CC);
	zend_declare_class_constant_long(phalcon_db_ce, SL("FETCH_OBJ"),         (long int)PDO_FETCH_OBJ TSRMLS_CC);
	zend_declare_class_constant_long(phalcon_db_ce, SL("FETCH_BOUND"),       (long int)PDO_FETCH_BOUND TSRMLS_CC);
	zend_declare_class_constant_long(phalcon_db_ce, SL("FETCH_COLUMN"),      (long int)PDO_FETCH_COLUMN TSRMLS_CC);
	zend_declare_class_constant_long(phalcon_db_ce, SL("FETCH_CLASS"),       (long int)PDO_FETCH_CLASS TSRMLS_CC);
	zend_declare_class_constant_long(phalcon_db_ce, SL("FETCH_INTO"),        (long int)PDO_FETCH_INTO TSRMLS_CC);
	zend_declare_class_constant_long(phalcon_db_ce, SL("FETCH_FUNC"),        (long int)PDO_FETCH_FUNC TSRMLS_CC);
	zend_declare_class_constant_long(phalcon_db_ce, SL("FETCH_NAMED"),       (long int)PDO_FETCH_NAMED TSRMLS_CC);
	zend_declare_class_constant_long(phalcon_db_ce, SL("FETCH_KEY_PAIR") ,   (long int)PDO_FETCH_KEY_PAIR TSRMLS_CC);

	zend_declare_class_constant_long(phalcon_db_ce, SL("FETCH_GROUP"),       (long int)PDO_FETCH_GROUP TSRMLS_CC);
	zend_declare_class_constant_long(phalcon_db_ce, SL("FETCH_UNIQUE"),      (long int)PDO_FETCH_UNIQUE TSRMLS_CC);
	zend_declare_class_constant_long(phalcon_db_ce, SL("FETCH_CLASSTYPE"),   (long int)PDO_FETCH_CLASSTYPE TSRMLS_CC);
	zend_declare_class_constant_long(phalcon_db_ce, SL("FETCH_SERIALIZE"),   (long int)PDO_FETCH_SERIALIZE TSRMLS_CC);
	zend_declare_class_constant_long(phalcon_db_ce, SL("FETCH_PROPS_LATE"),  (long int)PDO_FETCH_PROPS_LATE TSRMLS_CC);
</pre>

Phalcon 2.0
<pre>
	const FETCH_ASSOC = 1;
	const FETCH_BOTH = 2;
	const FETCH_NUM = 3;
	const FETCH_OBJ = 4;
</pre>

**Comment by sergeyklay** [comment]
Hm.. good reason for pull request :santa:

**Comment by salarmehr** [comment]
@olivier-monaco Actually, I want use `FETCH_COLUMN` because it returns an array instead of a resource.

**Comment by andresgutierrez** [comment]
This is fixed in Phalcon 2.0.4 (2.0.x branch)
